### PR TITLE
add prometheus metrics

### PR DESCRIPTION
**Pull request by mikemrm**

### CODE REVIEW

File: grpc_server.go [hunk 1]
```diff
@@ -86,11 +75,22 @@ func (s *server) Get(ctx context.Context, in *hegel.GetRequest) (*hegel.GetRespo
 }
 
 func (s *server) Subscribe(in *hegel.SubscribeRequest, stream hegel.Hegel_SubscribeServer) error {
-	active_subscription.Inc()
-	defer active_subscription.Dec()
+	metrics.TotalSubscriptions.Inc()
+	metrics.Subscriptions.WithLabelValues("initializing").Inc()
+	initError := func(err error) error {
+		s.log.Error(err)
```
Comment (mmlb) [inline grpc_server.go:81]: would be nice to have the ip address logged too once its been extracted from the request

Comment (mikemrm) [reply grpc_server.go:81]: Good point

File: grpc_server.go [hunk 2]
```diff
@@ -86,27 +75,43 @@ func (s *server) Get(ctx context.Context, in *hegel.GetRequest) (*hegel.GetRespo
...
+	logger := s.log.With("op", "subscribe")
+
+	initError := func(err error) error {
+		logger.Error(err)
+		metrics.Subscriptions.WithLabelValues("initializing").Dec()
+		labels := prometheus.Labels{
+			"op":    "subscribe",
+			"state": "initializing",
+		}
+		metrics.Errors.With(labels).Inc()
```
Comment (mmlb) [inline grpc_server.go:90]: why the `labels` var? Don't really need to change if you don't want though.

Comment (mikemrm) [reply grpc_server.go:90]: ah, you're right. Would be nice to keep it consistent.

**Pull request merged**